tests/migration_engine/test_db_inspectors.py

```diff
@@ -0,0 +1,204 @@
+import os
+from pathlib import Path
+import shutil
+import pytest
+from skibidi_orm.migration_engine.db_config.sqlite3_config import SQLite3Config
+from skibidi_orm.migration_engine.db_inspectors.db_inspector import SqliteInspector
+import sqlite3
+
+
+temp_dir = "./tmp"
+temp_db_file = "./tmp/test_db_inspectors.db"
+sql_table1 = """
+    CREATE TABLE table1 (
+        id INTEGER PRIMARY KEY,
+        name TEXT NOT NULL
+    );
+"""
+sql_table2 = """
+    CREATE TABLE table2 (
+        id INTEGER PRIMARY KEY,
+        name TEXT NOT NULL
+    );
+"""
+sql_table_primary_key_not_null = """
+    CREATE TABLE table_primary_key_not_null (
+        id INTEGER PRIMARY KEY NOT NULL
+    );
+"""
+
+sql_schema_with_fks = [
+    """
+    CREATE TABLE users (
+        user_id INTEGER PRIMARY KEY,
+        username TEXT NOT NULL UNIQUE,
+        email TEXT NOT NULL UNIQUE,
+        password_hash TEXT NOT NULL,
+        registration_date TIMESTAMP DEFAULT CURRENT_TIMESTAMP
+    );
+""",
+    """
+    CREATE TABLE posts (
+        post_id INTEGER PRIMARY KEY,
+        user_id INTEGER NOT NULL,
+        title TEXT NOT NULL,
+        content TEXT NOT NULL,
+        post_date TIMESTAMP DEFAULT CURRENT_TIMESTAMP,
+        FOREIGN KEY (user_id) REFERENCES users(user_id)
+    );
+""",
+    """
+    CREATE TABLE comments (
+        comment_id INTEGER PRIMARY KEY,
+        user_idd INTEGER NOT NULL,
+        post_id INTEGER NOT NULL,
+        comment_text TEXT NOT NULL,
+        comment_date TIMESTAMP DEFAULT CURRENT_TIMESTAMP,
+        FOREIGN KEY (user_idd) REFERENCES users(user_id),
+        FOREIGN KEY (post_id) REFERENCES posts(post_id)
+    );
+""",
+]
+
+
+def create_temp_db_file(temp_dir: str, db_file: str):
+    if os.path.exists(temp_dir):
```

Comment:
i think there's already a fixture for that with context manager syntax which is perfect for this use case